docs/hooks/hook_civicrm_alterMailingRecipients.md

```diff
@@ -0,0 +1,42 @@
+# hook_civicrm_alterMailingRecipients
+
+## Summary
+
+This hook is called to allow the user to alter the mailing recipients after they have been constructed.
+
+### Notes
+
+!!! Note
+    This hook is called two times, once at the start and once at the end of mail recipients
+    building, identified by $context as 'pre' or 'post' respectively
+
+## Availability
+
+This hook was first available in CiviCRM 4.7.32
+
+## Definition
+
+```php
+    function hook_civicrm_alterMailingRecipients(&$mailingObject, &$params, $context);
+```
+
+## Parameters
+
+-   object `$mailingObject` - reference to CRM_Mailing_DAO_Mailing object
+-   array `&$params` - the mailing parameters used to filter mailing recipients
+-   string `$context` - contain 'pre' or 'post' value to indicate when the hook is fired
```

Comment:
In the `$context`, I like how it provides concrete examples.

I think it's good do that with all array and string parameters -- so, *what keys and values are expected in `$params`*?

Reply:
+1 re providing more info on possible values in $params here. @monish when you're back next week can you find an authoritative source for content of $params...likely where they are consumed or perhaps marshalled somewhere in a complete UT?